### ISSUE TITLE
Getting Accuracy from validation results for CLASSIFICATION

### ISSUE THREAD

**Issue by james-imi** [ultralytics/ultralytics#14032]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I've been given this. What I want is that accuracy for validation

```
fitness: 0.6800000071525574
keys: ['metrics/accuracy_top1', 'metrics/accuracy_top5']
results_dict: {'metrics/accuracy_top1': 0.36000001430511475, 'metrics/accuracy_top5': 1.0, 'fitness': 0.6800000071525574}
```

**Comment by glenn-jocher** [comment]
@james-imi hello,

Thank you for reaching out with your question!

To obtain the accuracy from your validation results for a classification task, you can directly use the values from the `results_dict` that you have provided. Specifically, the `metrics/accuracy_top1` and `metrics/accuracy_top5` keys contain the top-1 and top-5 accuracy metrics, respectively.

Here's a quick breakdown of your results:
- **Top-1 Accuracy**: `metrics/accuracy_top1` is `0.36000001430511475`, which means that the model's top-1 accuracy is approximately 36%.
- **Top-5 Accuracy**: `metrics/accuracy_top5` is `1.0`, indicating that the model's top-5 accuracy is 100%.

If you need to extract and print these values programmatically, you can do so as follows:

```python
results_dict = {'metrics/accuracy_top1': 0.36000001430511475, 'metrics/accuracy_top5': 1.0, 'fitness': 0.6800000071525574}

top1_accuracy = results_dict['metrics/accuracy_top1']
top5_accuracy = results_dict['metrics/accuracy_top5']

print(f"Top-1 Accuracy: {top1_accuracy * 100:.2f}%")
print(f"Top-5 Accuracy: {top5_accuracy * 100:.2f}%")
```

This will output:
```
Top-1 Accuracy: 36.00%
Top-5 Accuracy: 100.00%
```

If you haven't already, please ensure you are using the latest version of the Ultralytics packages to avoid any potential issues that might have been resolved in recent updates.

For more detailed information on how to interpret these metrics or if you encounter any further issues, you can refer to our [documentation](https://docs.ultralytics.com).

Feel free to reach out if you have any more questions. Happy coding! 😊

**Comment by dsinha12345** [comment]
@glenn-jocher 

Hi there,
I just want to clarify, so the top1 and top5 accuracy values are obtained from the validation images and not the train images? If yes, is there any way I can get train accuracy of the model.

**Comment by glenn-jocher** [comment]
Hi @dsinha12345,

Yes, the `top1` and `top5` accuracy values you mentioned are indeed obtained from the validation images. To get the training accuracy, you can modify the training script to evaluate the model on the training dataset at the end of each epoch.

Here's a quick example of how you can do this using the Ultralytics YOLOv8 API:

```python
from ultralytics import YOLO

# Load the model
model = YOLO("yolov8n.pt")

# Train the model
results = model.train(data="coco128.yaml", epochs=10)

# Evaluate on the training dataset
train_results = model.val(data="coco128.yaml", split='train')

# Print training accuracy
print(f"Training Top-1 Accuracy: {train_results.box.map50 * 100:.2f}%")
print(f"Training Top-5 Accuracy: {train_results.box.map75 * 100:.2f}%")
```

This script will train the model and then evaluate it on the training dataset to get the training accuracy.

Feel free to reach out if you have more questions! 😊

**Comment by dsinha12345** [comment]
@glenn-jocher 

I'm sorry I think I didn't frame my question properly. I want to get a plot of training accuracy vs epochs. In the results.png, the plot is for Top_1 accuracy vs epochs which is essentially validation accuracy vs epochs as you clarified in your previous post. So is there any way I can get a training accuracy vs epochs curve?

**Comment by glenn-jocher** [comment]
Hi @dsinha12345,

To plot training accuracy vs. epochs, you can modify your training script to log the training accuracy at the end of each epoch. This can be done by evaluating the model on the training dataset during training. You can then use these logged values to create your plot.

If you need further assistance with this, please ensure you are using the latest version of the Ultralytics package. For more detailed guidance, you can refer to the Ultralytics documentation.
